### PR TITLE
Add snow simple flow e2e test

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#1108 

*Description of changes:*

Add simple flow e2e test for snow
- TestSnowKubernetes121SimpleFlow (create and delete w K8s version 1.21)

*Testing (if applicable):*

- `make local-e2e`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/framework/snow.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func (s *Snow) ClusterConfigFillers() []api.ClusterFiller {
+	ip, err := GenerateUniqueIp(s.cpCidr)
+	if err != nil {
+		s.t.Fatalf("failed to generate control plane ip for snow [cidr=%s]: %v", s.cpCidr, err)
+	}
+	s.clusterFillers = append(s.clusterFillers, api.WithControlPlaneEndpointIP(ip))
+
+	if s.podCidr != "" {
+		s.clusterFillers = append(s.clusterFillers, api.WithPodCidr(s.podCidr))
+	}
```
Comment (g-gaston) [inline test/framework/snow.go:75]: Maybe we should do all of this in the constructor?

**Review by g-gaston — COMMENTED** [review]
/lgtm

File: test/framework/snow.go (hunk 1 above)
Comment (jiayiwang7) [reply test/framework/snow.go:75]: i had the same thought and i think we should just deprecate `ClusterConfigFillers()`

**Pull request merged**